src/appleseed.shaders/src/appleseed/as_subsurface.osl

```diff
@@ -315,74 +343,89 @@ shader as_subsurface
 
     normal Nn = normalize(in_bump_normal);
 
-    if (in_sss_amount * max(in_color) > 0.0)
-    {
-        color albedo = clamp(in_sss_amount * in_color, 0.01, 0.99);
+    color albedo = clamp(in_sss_amount * in_color, 0.01, 0.99);
 
-        if (sss_profile == "randomwalk")
+    if (sss_profile == "randomwalk")
+    {
+        if (in_specular_weight > 0.0)
         {
-            out_outColor += as_subsurface(
-                sss_profile,
-                Nn,
-                albedo,
-                in_sss_mfp_scale * in_sss_mfp,
-                in_ior,
-                "fresnel_weight", in_fresnel_weight);
+            out_outColor += in_specular_weight *
+                as_randomwalk_glass(
+                    Nn,
+                    albedo,
+                    in_sss_mfp_scale * in_sss_mfp,
+                    in_ior,
+                    in_specular_roughness,
+                    "fresnel_weight", 1.0,
+                    "volume_anisotropy", in_volume_anisotropy);
         }
-        else
+        if (in_specular_weight < 1.0)
         {
-            out_outColor += as_subsurface(
-                sss_profile,
-                Nn,
-                albedo,
-                in_sss_mfp_scale * in_sss_mfp,
-                in_ior,
-                "fresnel_weight", in_fresnel_weight);
+            out_outColor += (1.0 - in_specular_weight) *
+                as_subsurface(
```

Comment:
Maybe add a comment explaining that you're computing random-walk SSS with a _diffuse_ interface here.

Reply:
Though it's an interesting idea to blend between the diffuse and glassy random walk with the specular term, it will restrict the ability to layer the GGX lobe over the diffuse random walk. Adding a parameter *transmission* to linearly blend from the *diffuse* to *glassy* random walk would be a possibility, but i'm not sure this is desirable at this stage, so we have instead the explicit *diffuse* random walk as previously, and a new *glassy* one. I think this removes the need for the comment here.